server/controllers/changeNonResidentialType/details.ts

```diff
@@ -51,9 +51,16 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
       const token = await req.services.authService.getSystemClientToken(user.username)
       const { values } = req.form
       const preSelectedConvertedCellType = res.locals.location.raw?.convertedCellType || []
+      const preOtherTypeChanged = res.locals.location.raw?.otherConvertedCellType || []
+
       const selectedConvertedCellType = values.convertedCellType
+      const selectedOtherConvertedCellType = values.otherConvertedCellType
+
       const isSameAsPreSelected = preSelectedConvertedCellType.includes(selectedConvertedCellType)
+      const isOtherTypeChanged = preOtherTypeChanged.includes(selectedOtherConvertedCellType)
+
       req.sessionModel.set('convertedCellTypeChanged', !isSameAsPreSelected)
+      req.sessionModel.set('otherTypeChanged', !isOtherTypeChanged)
```

Comment:
This is done a bit differently to most of the other journeys. In the other journeys we use the `validate` method to check if the value has changed and if it hasn't then we just redirect back to the view location page before it gets to `saveValues` or `successHandler`. That way we don't hit the API for no reason. But it was already done this way and you are just adding to it so it's ok for now.

Reply:
@marcusaleman 
https://github.com/ministryofjustice/hmpps-locations-inside-prison/blob/main/server/controllers/changeUsedFor/details.ts#L67-L74